### PR TITLE
Incorporate NASA-SPIRE data using SATID

### PR DESCRIPTION
**Pull request by mohar123**
Changed the prepbykx.f90 to extract observations based on satid value. This change is made to extract NASA-SPIRE from the prepbufr files based on their SATIDs rather than their KX values.

### CODE REVIEW

**Review by gmao-jjin3 — DISMISSED** [review]

**Comment by gmao-msienkie** [comment]
As I have mentioned before, I am not in favor of modifying/using 'prepbykx.f' for processing anything but PREPBUFR format data.  I feel like other data formats should have a separate program, mainly because most of the BUFR mnemonics it is trying to read are not present in other files.  If there is a separate program for extracting data by SATID it could be enhanced to also include subsetting by latitude/longitude (CLATH/CLONH not YOB/XOB) or possibly other mnemonics that are unique to the GNSSRO.

**Comment by rtodling** [comment]
Mohar, I am gong to agree w/ Meta on this ... perhaps you can simply rename this program into a new name - and revive the original prepbufr program ....

**Comment by mohar123** [comment]
I am going to rename the code and commit it but I would need help to trace my way back from the git push that I did from my working directory. If I change the name of the code to a brand new addition, I would require changing the Makefile as well, right?

File: GEOSaana_GridComp/GSI_GridComp/CMakeLists.txt [hunk 1]
```diff
@@ -700,6 +701,8 @@ if (CMAKE_Fortran_COMPILER_ID MATCHES Intel)
             set_source_files_properties (${src} PROPERTIES COMPILE_FLAGS "")
          elseif (${src} MATCHES prepbykx.f)
             set_source_files_properties (${src} PROPERTIES COMPILE_FLAGS "${GEOS_Fortran_Release_Flags} ${common_Fortran_flags} ${BYTERECLEN} ${EXTENDED_SOURCE} ${FP_MODEL_STRICT} ${ALIGNCOM}")
+         elseif (${src} MATCHES prepbysaid.f)
+            set_source_files_properties (${src} PROPERTIES COMPILE_FLAGS "${GEOS_Fortran_Release_Flags} ${common_Fortran_flags} ${BYTERECLEN} ${EXTENDED_SOURCE} ${FP_MODEL_STRICT} ${ALIGNCOM}")
```
Comment (tclune) [inline GEOSaana_GridComp/GSI_GridComp/CMakeLists.txt:705]: Why do you want to force the release flags?    Would be nice to be able to use debug flags when intended?

Comment (mohar123) [reply GEOSaana_GridComp/GSI_GridComp/CMakeLists.txt:705]: Hi Tom,

I have simply copied from the code above "prepbykx.f". Not sure about the nuances in the CMakefile. Please let me know what I should change it to.

-Mohar

Comment (rtodling) [reply GEOSaana_GridComp/GSI_GridComp/CMakeLists.txt:705]: @mohar123 and @tclune I will approave this and resolve the issue w/ the flag elsewhere.

**Review by rtodling — DISMISSED** [review]
I will resolve compile flag elsewhere.

**Review by rtodling — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**